### PR TITLE
Adds passes for (pad + consumer) to CPU pipeline and enables benchmarks.

### PR DESCRIPTION
**Pull request by hanhanW**
The benchmarks are tracked under experimental-flags.

### CODE REVIEW

**Comment by dcaballe** [comment]
Thanks, Hanhan! Could you please elaborate on why we have to limit the fusion transformation to x86 CPUs and we can't apply it generally to all the backends?

**Comment by MaheshRavishankar** [comment]
The regressions seem real to me?

**Comment by MaheshRavishankar** [comment]
> Thanks, Hanhan! Could you please elaborate on why we have to limit the fusion transformation to x86 CPUs and we can't apply it generally to all the backends?

Ideally the default should be fusion with producer. We have all the pieces to plumb that through, but needs to be owned by someone to connect it end to end.

**Comment by hanhanW** [comment]
> The regressions seem real to me?

Weird.. I thought all the regression was caught and addressed. I did look into the DeepLab case, let me take a look at it.

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 1]
```diff
@@ -441,6 +448,14 @@ void addMultiTilingExpertPassPipeline(OpPassManager &passManager,
     }
   }
 
+  {
+    nestedModulePM.addNestedPass<func::FuncOp>(
+        createFuseTensorPadWithConsumerPass());
+    nestedModulePM.addNestedPass<func::FuncOp>(
+        createConcretizePadResultShapePass());
+    nestedModulePM.addNestedPass<func::FuncOp>(createVectorizePadPass());
+  }
```
Comment (dcaballe) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp:457]: extra scope is not needed here and below?

**Review by dcaballe — APPROVED** [review]
LGTM!

File: compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp (hunk 1 above)
Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp:457]: good point!

**Review by mariecwhite — APPROVED** [review]

**Comment by pzread** [comment]
I double checked the benchmark flags, are we missing the `"--iree-flow-enable-fuse-padding-into-linalg-consumer-ops"` for ARMv8 local-sync benchmark? (line 255)

**Comment by hanhanW** [comment]
> I double checked the benchmark flags, are we missing the `"--iree-flow-enable-fuse-padding-into-linalg-consumer-ops"` for ARMv8 local-sync benchmark? (line 255)

Yes, you're right. Good catch..

**Pull request merged**